inference-engine/thirdparty/clDNN/src/program.cpp

```diff
@@ -1124,13 +1124,13 @@ void program::dump_program(const char* stage,
 
 data_types program::get_inference_precision(const program_node& node) const {
     if (node.is_input()) {
-        return node.get_output_layout().data_type;
+        return const_cast<program_node&>(node).get_output_layout().data_type;
     }
     std::vector<data_types> input_dts;
     for (auto& dep : node.get_dependencies()) {
         input_dts.push_back(dep->get_output_layout().data_type);
     }
-    data_types output_dt = node.get_output_layout().data_type;
+    data_types output_dt = const_cast<program_node&>(node).get_output_layout().data_type;
```

Comment:
Have you checked why node layout is not calculated at this point? If we keep this code, then dumps may impact final graph representation comparing to original one. So I'd prefer it to be fixed somewhere in graph optimizer passes.

Reply:
Node layout was already being calculated here.
The problem was caused by calling 'program_node::get_output_layout() const' instead of 'program_node::get_output_layout(bool)' as the function argument was changed from 'program_node&' to 'const program_node&'.
Do you want the node layout to be calculated somewhere else?
The first bad commit is 7a16de290aca623f8ed377b30775681c7458b199.
[GPU] Fixed NHWC interp. Added exec precision to impl name in perf counter

Reply:
I've updated it again after discussion with Mingyu. Could you review again?